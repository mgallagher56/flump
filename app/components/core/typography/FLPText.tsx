import React from 'react';

import type { TextProps } from '@chakra-ui/react';
import { Text } from '@chakra-ui/react';

const FLPText: React.FC<TextProps> = ({
  fontSize = 'md',
  fontWeight = 'normal',
  lineHeight = 'base',
  letterSpacing = 'normal',
  ...props
}) => {
  return (
    <Text fontSize={fontSize} fontWeight={fontWeight} lineHeight={lineHeight} letterSpacing={letterSpacing} {...props}>
      {props.children}
    </Text>
  );
};

export default FLPText;
