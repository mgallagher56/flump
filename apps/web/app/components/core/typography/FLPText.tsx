import type { FC } from 'react';

import type { TextProps } from '@chakra-ui/react';
import { Text } from '@chakra-ui/react';

const FLPText: FC<TextProps> = ({
  fontSize = 'md',
  fontWeight = 'normal',
  lineHeight = 'base',
  letterSpacing = 'normal',
  ...props
}) => {
  return (
    <Text fontSize={fontSize} fontWeight={fontWeight} letterSpacing={letterSpacing} lineHeight={lineHeight} {...props}>
      {props.children}
    </Text>
  );
};

export default FLPText;
