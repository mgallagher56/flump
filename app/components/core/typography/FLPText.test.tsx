import { render } from '@testing-library/react';

import FLPText from './FLPText';

const sampleText =
  'Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.';

describe('FLPText', () => {
  test('renders as expected', () => {
    const { container } = render(<FLPText>{sampleText}</FLPText>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a fontSiez', () => {
    const { container } = render(<FLPText fontSize="20">{sampleText}</FLPText>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a fontWeight', () => {
    const { container } = render(<FLPText fontWeight="bold">{sampleText}</FLPText>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a lineHeight', () => {
    const { container } = render(<FLPText lineHeight={40}>{sampleText}</FLPText>);
    expect(container).toMatchSnapshot();
  });
  test('renders as expected with a letterSpacing', () => {
    const { container } = render(<FLPText letterSpacing="widest">{sampleText}</FLPText>);
    expect(container).toMatchSnapshot();
  });
});
