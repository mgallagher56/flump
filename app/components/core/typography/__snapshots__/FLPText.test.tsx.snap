// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FLPText > renders as expected 1`] = `
<div>
  <p
    class="chakra-text css-km77ca"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </p>
</div>
`;

exports[`FLPText > renders as expected with a fontSiez 1`] = `
<div>
  <p
    class="chakra-text css-ujtdc1"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </p>
</div>
`;

exports[`FLPText > renders as expected with a fontWeight 1`] = `
<div>
  <p
    class="chakra-text css-hmjgxl"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </p>
</div>
`;

exports[`FLPText > renders as expected with a letterSpacing 1`] = `
<div>
  <p
    class="chakra-text css-w04h5u"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </p>
</div>
`;

exports[`FLPText > renders as expected with a lineHeight 1`] = `
<div>
  <p
    class="chakra-text css-dxclju"
  >
    Lorem, ipsum dolor sit amet consectetur adipisicing elit. Nemo quaerat repellendus numquam et esse ut labore molestias nihil optio expedita atque, quas eligendi corrupti quasi impedit suscipit sunt.
  </p>
</div>
`;
